### DISCUSSION TITLE
Updating instance on DigitalOcean droplet

### DISCUSSION THREAD

**Discussion by JammyPuffin** [inventree/InvenTree#3766] [Q&A] [answered]
I am using windows. I have installed InvenTree using the one click install through Digital Ocean and the instillation was successful however the droplet that was created uses the 0.6.3 version of InvenTree. I would like to update it to the current version as well as keep up to date when new versions are released. I am not too well versed in how to go about doing this. From what I read in the documentation the webserver needs to be stopped while pulling the new version however I'm not sure how to stop this without also stopping the Digital Ocean droplet which stops me from accessing the terminal to perform the pull.
Any help would be much appreciated.

**Comment by SchrodingersGat** [comment]
@matmair do you have any suggestions here?

**Comment by matmair** [comment, marked as answer]
@JammyPuffin the droplet is just regular Ubuntu. Stop the following services with `sudo service ${name} stop`:
- nginx
- inventree-web
- inventree-cluster

**Reply by JammyPuffin** [reply]
Thank you for the fast reply. I have been experimenting which is why it took me long to get back to you.
So once the server is started I navigate to the repository location and stop the services you mentioned. 
![Capture1](https://user-images.githubusercontent.com/115537591/195094725-2693ed04-c920-4275-aacc-c82a94041ce6.PNG)

Once I confirm these have stopped I perform git pull origin master. It asks for some credentials which I enter, then restart the git pull. This throws out a tone of conflicts.
![Capture2](https://user-images.githubusercontent.com/115537591/195095292-d49c5d85-fb05-42ee-96ea-788f38925ccb.PNG)

After this the services cannot be restarted. They keep crashing.
![Capture3](https://user-images.githubusercontent.com/115537591/195095651-af2d1855-44da-4291-85fb-dc5792c0f0e3.PNG)

The only way I can get the server functioning again is by destroying and remaking the droplet.

What should the sequence of the update be? I tried many ways but all failed.
Thank you for your help.

**Reply by matmair** [reply]
@JammyPuffin can you confirm it pulled master/stable? The best way would be to post the response to `git log -n 1` in `/home/inventree/src`.
Please try to manually start the server and look at the log output. For this you should:
- move to inventree's home
- activate venv
- start up the server with `invoke server`
What does the system log out there?

**Reply by JammyPuffin** [reply]
When I do git log -n 1 in /home/inventree/src I get this.
![Capture4](https://user-images.githubusercontent.com/115537591/195538101-c3299a12-c2ef-4f02-87ba-96a6f17bc40f.PNG)
Then activating venv and using the invoke server command I get this.
![Capture6](https://user-images.githubusercontent.com/115537591/195539478-29caf2d4-2ff0-42bc-84d6-bb2f77c95120.PNG)

**Reply by matmair** [reply]
@JammyPuffin can you check for an env in `/home/inventree/`?

**Reply by JammyPuffin** [reply]
@matmair yep that seems to be there.
![Capture7](https://user-images.githubusercontent.com/115537591/195794324-adcdf4f1-42aa-4364-b042-eced180b7936.PNG)

**Reply by matmair** [reply]
@JammyPuffin did you create venv? Normally there should only be env

**Reply by matmair** [reply]
@JammyPuffin do you need further help or can we close this as answered?

**Reply by JammyPuffin** [reply]
Thank you you can close this thread. :)
